### ISSUE TITLE
MIDI control bank change fail

### ISSUE THREAD

**Issue by simdax** [amsynth/amsynth#35]
Hello,

Not really a bug, but I'm trying to communicate between AMSYNTH and SUPERCOLLIDER, sending simple MIDI control Message.

It works for volume or program change but I can't change bank ??

I was looking sources and saw that

```
void
MidiController::controller_change(unsigned char cc, unsigned char value)
{
	if (!_handler || !presetController)
		return;

	switch (cc) {
		case MIDI_CC_BANK_SELECT_LSB:
		case MIDI_CC_BANK_SELECT_MSB:
break;
```

Maybe it's not implemented yet, or am I missing something ? Should be difficult to implement ?

Thank you !

**Comment by nixxcode** [comment]
Hi @simdax - yes, bank selection via MIDI is not currently implemented.

To do so would require assigning numbers to each bank, and right now I'm not sure how best to do that. As a user, how would you expect that to work?

**Comment by simdax** [comment]
As for an example, with ZynAddSubFx and Supercollider, I just send a midiControl #0 and I can choose the bank.

In amsynth, I can imagine that [U] User bank should be 0, all the [F] BrianBanks in the range (1..19) and 20 for amsynth factory. This are the banks I have.

**Comment by nickdowell** [comment]
@phil123456 what version of amsynth are you using?

MIDI Bank Selection was added in [1.7.0](https://github.com/amsynth/amsynth/releases/tag/release-1.7.0)

**Comment by phil123456** [comment]
version 1.12.2...

I gave it another try
I sent this :
[B0 00 00]
[B0 20 02]
[C0 00]

amsynth BANK does not move an inch...it stays on "[U] user bank"

while program changes are taken into account

**Comment by nickdowell** [comment]
Hmm ok I think there's a bug with amsynth's MIDI bank select implementation - the LSB and MSB messages are the [wrong way round](https://github.com/amsynth/amsynth/blob/develop/src/MidiController.cpp#L166-L173). It's ended up with way because amsynth was tested against vkeybd which [sends the bank in CC0](https://github.com/tiwai/vkeybd/blob/master/oper_alsa.c#L271-L284) 🙈

**Comment by nickdowell** [comment]
There's now a fix for the bank select handling in #190 on branch `feature/fix-midi-bank-select`

Would be very helpful if someone can report whether the fix works for them 🙃
